ignite/contrib/handlers/lr_finder.py

```diff
@@ -256,14 +269,22 @@ def plot(
 
     def lr_suggestion(self) -> Any:
         """
-        Returns: learning rate at the minimum numerical gradient
+        Returns: the learning rate at the minimum numerical gradient
+        (ignores the increasing part of the curve)
         """
         if not self._history:
             raise RuntimeError("learning rate finder didn't run yet so lr_suggestion can't be returned")
         loss = self._history["loss"]
-        grads = torch.tensor([loss[i] - loss[i - 1] for i in range(1, len(loss))])
+        min_loss_idx = torch.tensor(loss).argmin()
+        # Ignore the increasing part of the curve
+        decreasing_losses = self._history["loss"][: int(min_loss_idx.item()) + 1]
+        if len(decreasing_losses) == 1:
+            raise RuntimeError("FastaiLRFinder got unexpected curve shape, the curve should be somehow U-shaped")
+        losses = torch.tensor(decreasing_losses)
+        grads = torch.tensor([losses[i] - losses[i - 1] for i in range(1, len(losses))])
         min_grad_idx = grads.argmin() + 1
-        return self._history["lr"][int(min_grad_idx)]
+        self._min_grad_idx = int(min_grad_idx)
+        return self._history["lr"][self._min_grad_idx]
```

Comment:
I think we can keep `_min_grad_idx` as a local var `min_grad_idx` instead of instance attribute `self._min_grad_idx`.